transforms/universal/opensearch/dpk_opensearch/transform.py

```diff
@@ -11,199 +11,292 @@
+default_username = "admin"
+default_port = "9200"
+default_docid_column_name = "document_id"
+default_embeddings_column_name = "embeddings"
+default_content_column_name = "contents"
+default_filename = "filename"
+default_delete_index = False
+user = os.environ.get("OPENSEARH_USERID", "admin")
 
 
-class OpenSearchTransform(AbstractTableTransform):
-    """
-    Implements a simple copy of a pyarrow Table.
-    """
+class OpenSearchTransform(AbstractTableTransform, SinkHandler):
 
-    def __init__(self, config: dict[str, Any]={}):
-        """
-        Initialize based on the dictionary of configuration information.
-        This is generally called with configuration parsed from the CLI arguments defined
-        by the companion runtime, NOOPTransformRuntime.  If running inside the RayMutatingDriver,
-        these will be provided by that class with help from the RayMutatingDriver.
-        """
-        # Make sure that the param name corresponds to the name used in apply_input_params method
-        # of NOOPTransformConfiguration class
+    def __init__(self, config: dict[str, Any]):
         super().__init__(config)
-        x=config.get(host_cli_param, host_default).split(':')
+        self.logger = get_logger(__name__)
+
+        x = config.get(hostname, default_host).split(':')
         self.host = x[0]
-        self.port = x[1] if len(x) > 1 else port_default
-        self.docId = config.get(docid_cli_param, doc_id_default)
-        self.ndx= config.get(ndx_cli_param, 
-                               f"dpk_{datetime.now().strftime('%y%m%d%H%M%S')}")
+        self.port = x[1] if len(x) > 1 else default_port
+        self.doc_id_column = config.get(docid_column_name_key, default_docid_column_name)
 
-        self.uid=os.environ.get("OPENSEARH_USERID", "admin")
+        self.index_name = config.get(indx, f"dpk_{datetime.now().strftime('%y%m%d%H%M%S')}")
+        self.embeddings_column = config.get(embeddings_column_name_key, default_embeddings_column_name)
+        self.content_column = config.get(content_column_name_key, default_content_column_name)
+        self.dimension_size = config.get(dimension_size)
+        self.delete_index = config.get(delete_index, default_delete_index)
+
+        self.uid = user
         try:
-            self.pwd=os.environ["OPENSEARCH_PASSWORD"]
+            self.pwd = os.environ["OPENSEARCH_PASSWORD"]
         except KeyError as e:
-            logger.error(f"Environment variable OPENSEARCH_PASSWORD must be define. Raising Exception: {e}")
+            self.logger.error(f"Environment variable OPENSEARCH_PASSWORD must be define. Raising Exception: {e}")
             raise UnrecoverableException("Missing credentials")
-
+        try:
+            self.client = OpenSearch(
+                hosts=[{'host': self.host, 'port': self.port}],
+                http_compress=True,  # enables gzip compression for request bodies
+                http_auth=(self.uid, self.pwd),
+                use_ssl=True,
```

Comment:
We configure the OpenSearch client to work over SSL, but it is not always correct. If security on the OpenSearch is disabled, it will listen over HTTP only.